commerce_coordinator/apps/commercetools/views.py

```diff
@@ -34,12 +44,18 @@ def post(self, request):
 
         logger.debug('[CT-OrderFulfillView] Message received from commercetools with details: %s', input_data)
 
-        message_details = OrderFulfillMessageInputSerializer(data=input_data)
+        message_details = OrderMessageInputSerializer(data=input_data)
         message_details.is_valid(raise_exception=True)
 
         client = CommercetoolsAPIClient()
         order_id = message_details.data['order_id']
-        order = client.get_order_by_id(order_id)
+
+        try:
+            order = client.get_order_by_id(order_id)
+        except CommercetoolsError as _:  # pragma no cover
+            logger.error("Order not found: %s", order_id)
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
         customer = client.get_customer_by_id(order.customer_id)
```

Comment:
Suggestion:
- Please move `get_customer_by_id` into try/except block as well
- Change the logger function to exception as we need to be alerted on just exceptions
- Make exception message more meaningful and append API error as well

```
try:
    order = client.get_order_by_id(order_id)
    customer = client.get_customer_by_id(order.customer_id)
except CommercetoolsError as exc:
    logger.exception(f'Error fetching CT Customer or Order with ID "{order_id}". Error: {exc}')
    return Response(status=status.HTTP_404_NOT_FOUND)
```